stdpopsim/cli.py

```diff
@@ -509,10 +509,27 @@ def stdpopsim_cli_parser():
         choices=[e.id for e in stdpopsim.all_engines()],
         help="Specify a simulation engine.")
 
-    for engine in stdpopsim.all_engines():
-        group = top_parser.add_argument_group(
-                f"{engine.id} specific parameters")
-        engine.add_arguments(group)
+    msprime = stdpopsim.get_engine("msprime")
+    msprime_parser = top_parser.add_argument_group("msprime specific parameters")
+    msprime_parser.add_argument(
+                "--msprime-model",
+                default=msprime.supported_models[0],
+                choices=msprime.supported_models,
+                help="Specify the simulation model used by msprime. "
+                     "See msprime API documentation for details.")
+    msprime_parser.add_argument(
+                "--msprime-change-model",
+                default=[],
+                choices=msprime.supported_models, action="append",
```

Comment:
I thought it would be possible to set ``nargs=2`` here and so something like 
```
--msprime-change-model 10 dtwf --msprime-change-model 100 hudson
```
would give an args list [("10", "dtwf"), ("100", "hudson")]. Is it possible to mix nargs with ``action="append"``?

Reply:
Yeah, even the metavar works with nargs:
```
parser.add_argument('--msprime-change-model',
        metavar=("T","MODEL"),
        default=[], action="append", nargs=2)
```
But `type=some_func` doesn't, because argparse wants to call the converting function separately for each of the nargs. So we'll still need to have something like the following in stdpopsim_main() after calling parse_args() to check the types:

```
change_model = []
for t, model in args.msprime_change_model:
    try:
        t = float(t)
    except ValueError:
        parser.error(f"`{t}' is not a number")
    if model not in supported_models:
        parser.error(f"`{model}' is not a supported model")
    change_model.append((t, model))
args.msprime_change_model = change_model
```

Reply:
Actually, `type=some_func` can be made to work through the magic of mutable default function parameters...
```
def tuple_of_time_model(arg, _expect_time=[True,]):
    if _expect_time[0]:
        try:
            arg = float(arg)
        except ValueError:
            parser.error(f"`{arg}' is not a number")
    else:
        if arg not in supported_models:
            parser.error(f"`{arg}' is not a supported model")
    _expect_time[0] = not _expect_time[0]
    return arg

parser.add_argument('--msprime-change-model',
        metavar=("T","MODEL"),
        type=tuple_of_time_model,
        default=[], action="append", nargs=2)
```